pulpcore/cli/common/generic.py

```diff
@@ -217,11 +218,12 @@ def load_json_callback(
 
 
 def create_content_json_callback(
-    content_ctx: Type[PulpContentContext], schema: s.Schema = None
+    ctx_class: Optional[Type[PulpContentContext]] = None, schema: s.Schema = None
 ) -> Any:
     def _callback(
         ctx: click.Context, param: click.Parameter, value: Optional[str]
     ) -> Optional[List[PulpContentContext]]:
+        nonlocal ctx_class
```

Comment:
This I don't understand. I think this allows to overwrite the value in the closure (that is the scope created by calling `create_content_json_callback`. And I think, this scope may live longer than we expect here when using the `shell` command. So where I expect this to fail is: Inside `pulp shell` call a content modify command for collections and then one for roles. In the second call, we might have the `CollectionContext` still in the `ctx_class`. For single one shot commands, this is probably just not correct, but no problem.